src/lib.rs

```diff
@@ -90,41 +100,107 @@ pub struct IronOxide {
+}
+
 /// Initialize the IronOxide SDK with a device. Verifies that the provided user/segment exists and the provided device
 /// keys are valid and exist for the provided account. If successful returns an instance of the IronOxide SDK
 pub fn initialize(device_context: &DeviceContext) -> Result<IronOxide> {
-    // 1 MB
-    const BYTES_BEFORE_RESEEDING: u64 = 1 * 1024 * 1024;
     let mut rt = Runtime::new().unwrap();
-    let account_id = device_context.account_id();
-    rt.block_on(crate::internal::user_api::user_key_list(
+    rt.block_on(crate::internal::user_api::user_get_current(
         &device_context.auth(),
-        &vec![account_id.clone()],
     ))
-    .and_then(|mut users| {
-        users
-            //We're using remove here because we don't actually need this HashMap anymore and remove
-            //returns us ownership so we can avoid a clone.
-            .remove(&device_context.account_id())
-            .map(|current_user_public_key| IronOxide {
-                recrypt: Recrypt::new(),
-                device: device_context.clone(),
-                user_master_pub_key: current_user_public_key,
-                rng: Mutex::new(ReseedingRng::new(
-                    rand_chacha::ChaChaCore::from_entropy(),
-                    BYTES_BEFORE_RESEEDING,
-                    EntropyRng::new(),
-                )),
-            })
-            .ok_or(IronOxideErr::InitializeError)
+    .and_then(|current_user| {
+        Ok(IronOxide::create_with_side_effects(
+            &current_user,
+            device_context,
+        ))
     })
+    .map_err(|_| IronOxideErr::InitializeError)
+}
+
+/// Initialize the IronOxide SDK and check to see if the user that owns this `DeviceContext` is
+/// marked for private key rotation, or if any of the groups that the user is an admin of is marked
+/// for private key rotation.
+pub fn initialize_check_rotation(device_context: &DeviceContext) -> Result<InitAndRotationCheck> {
+    Runtime::new().unwrap().block_on(
+        internal::user_api::user_get_current(device_context.auth()).and_then(|curr_user| {
+            let ironoxide = IronOxide::create_with_side_effects(&curr_user, &device_context);
+
+            if curr_user.needs_rotation() {
+                Ok(InitAndRotationCheck::RotationNeeded(
+                    ironoxide,
+                    PrivateKeyRotationCheckResult {
+                        rotations_needed: EitherOrBoth::Left(curr_user.account_id().clone()),
+                    },
+                ))
+            } else {
+                Ok(InitAndRotationCheck::NoRotationNeeded(ironoxide))
+            }
+        }),
+    )
 }
 
 impl IronOxide {
     /// Get the `DeviceContext` instance that was used to create this SDK instance
     pub fn device(&self) -> &DeviceContext {
         &self.device
     }
+
+    fn create_with_side_effects(
```

Comment:
I don't understand this name.

Reply:
Meant to indicate that we are depending on RNG seeding to create this IronOxide. Could potentially be shortened to `create` -- `new` seems to have the connotation (to me) that the construction is pure.

Reply:
create with a comment seems better to me.